### PR TITLE
[Test-Proxy] Allow `x-recording-file` to POST body vs header

### PR DESCRIPTION
**Pull request by scbedd**
@JoshLove-msft here ya go.

Todo:

- [x] Update code and tests
- [x] Update docs
- [x] Update Samples

### CODE REVIEW

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/TransformTests.cs [hunk 1]
```diff
@@ -27,7 +27,10 @@ public async Task ConditionalTransformAppliesForRegex()
             var untransformedEntry = TestHelpers.LoadRecordSession(targetFile).Session.Entries[1];
 
             // start playback
-            playbackContext.Request.Headers["x-recording-file"] = targetFile;
+            var body = "{\"x-recording-file\":\"" + targetFile + "\"}";
```
Comment (JoshLove-msft) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/TransformTests.cs:30]: Can we add a test case with non-ascii chars so we make sure everything works as expected?

**Review by JoshLove-msft — APPROVED** [review]

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/PlaybackTests.cs [hunk 2]
```diff
@@ -20,7 +20,9 @@ public async void TestStartPlaybackSimple()
         {
             RecordingHandler testRecordingHandler = new RecordingHandler(Directory.GetCurrentDirectory());
             var httpContext = new DefaultHttpContext();
-            httpContext.Request.Headers["x-recording-file"] = "Test.RecordEntries/requests_with_continuation.json";
+            var body = "{\"x-recording-file\":\"Test.RecordEntries/requests_with_continuation.json\"}";
```
Comment (JoshLove-msft) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/PlaybackTests.cs:23]: Can we add a test case with non-ascii chars so we make sure everything works as expected?

Comment (scbedd) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/PlaybackTests.cs:23]: Yeah. Definitely.

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/RecordTests.cs [hunk 3]
```diff
@@ -50,21 +52,24 @@ public void TestStartRecordInMemory()
...
 
             var (fileName, session) = testRecordingHandler.RecordingSessions[recordingId];
 
             Assert.Empty(fileName);
         }
 
-        [Fact]
-        public void TestStopRecordingSimple()
+        [Theory]
+        [InlineData("recordings/TestStartRecordSimple.json")]
```
Comment (JoshLove-msft) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/RecordTests.cs:64]: Is this intentionally the same as the next test case?

Comment (scbedd) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/RecordTests.cs:64]: It's not though! 

```
Simple
vs
Simplé
```

Comment (JoshLove-msft) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/RecordTests.cs:64]: Ah I missed that 😆 
Thanks!

**Pull request merged**